orangecontrib/bio/widgets3/OWdictyExpress.py

```diff
@@ -0,0 +1,235 @@
+                                      callback=self.auth_changed)
+
+        self.passfield = gui.lineEdit(box, self, "password", "Password:", labelWidth=100, orientation='horizontal',
+                                      callback=self.auth_changed)
+
+        self.passfield.setEchoMode(QLineEdit.Password)
+
+        self.controlArea.layout().addWidget(h_line())
+
+        self.commit_button = gui.button(self.controlArea, self, "Commit", callback=self.commit)
+        self.handle_commit_button(False)
+
+        self.refresh_button = gui.button(self.controlArea, self, "Refresh", callback=self.refresh)
+        self.handle_cache_button(True)
+
+        gui.rubber(self.controlArea)
+
+        # Experiment Section
+
+        label = QLabel("Available projects:")
+        my_font = QFont()
+        my_font.setBold(True)
+        label.setFont(my_font)
+        self.mainArea.layout().addWidget(label)
+
+        self.mainArea.layout().addWidget(h_line())
+
+        self.filter = gui.lineEdit(self.mainArea, self, "searchString", "Filter:", callbackOnType=True,
+                                   callback=self.search_update)
+
+        self.experimentsWidget = QTreeWidget()
+        self.experimentsWidget.setAlternatingRowColors(True)
+        self.experimentsWidget.setRootIsDecorated(False)
+        self.experimentsWidget.setUniformRowHeights(True)
+        self.experimentsWidget.setSortingEnabled(True)
+
+        self.experimentsWidget.setItemDelegateForColumn(
+            0, gui.IndicatorItemDelegate(self, role=Qt.DisplayRole))
+
+        self.experimentsWidget.selectionModel().selectionChanged.connect(
+            self.onSelectionChanged)
+
+        self.experimentsWidget.setHeaderLabels(self.headerLabels)
+        self.mainArea.layout().addWidget(self.experimentsWidget)
+
+        if self.username and self.password:
+            self.connect()
+
+    def auth_set(self):
+        if len(self.username):
+            self.passfield.setDisabled(False)
+        else:
+            self.passfield.setDisabled(True)
+
+    def auth_changed(self):
+        self.auth_set()
+        self.connect()
+
+    def refresh(self):
+        self.reset()
+        self.load_experiments()
+
+    def reset(self):
+        self.experimentsWidget.clear()  # clear QTreeWidget
+        self.items = []
+        self.lastSelected = None
+        self.searchString = ""
+        self.handle_commit_button(False)
+
+    def search_update(self):
+        for item in self.items:
+            item.setHidden(
+                not all(s in item for s in self.searchString.split()))
+
+    def connect(self):
+        self.res = None
+        self.error(1)
+        self.reset()
+        self.handle_commit_button(False)
+        self.handle_cache_button(False)
+
+        if self.username and self.password:
+            try:
+                self.res = Genapi(self.username, self.password, self.server)
+            except Exception as e:
```

Comment:
You catch your expected specific `GenAPIExceptions` here. `Exception` is then reserved for something unexpected.